pkg/recorder/recorder.go

```diff
@@ -85,12 +93,33 @@ func (r *Recorder) Record(rec record.Record) error {
 	}
 	// we want to record our metadata file anyway
 	if r.size+recordSize > r.maxArchiveSize && rec.Name != MetadataRecordName {
-		return fmt.Errorf("record %s(size=%d) exceeds the archive size limit %d and will not be included in the archive",
-			recordName, recordSize, r.maxArchiveSize)
+		errs = append(errs, fmt.Errorf(
+			"record %s(size=%d) exceeds the archive size limit %d and will not be included in the archive",
+			recordName, recordSize, r.maxArchiveSize,
+		))
+		return
+	}
+
+	if existingRecord, found := r.records[rec.GetFilename()]; found {
+		errs = append(errs, fmt.Errorf(
+			`a record with the name "%v" was already recorded and had fingerprint "%v", `+
+				`overwriting with data having fingerprint "%v"`,
+			rec.GetFilename(), existingRecord.Fingerprint, fingerprint,
+		))
 	}
+	if existingPath, found := r.recordedFingerprints[fingerprint]; found {
+		// this doesn't necessarily mean it's an error. There can be a collision after hashing
+		errs = append(errs, &types.Warning{UnderlyingValue: fmt.Errorf(
+			`a record with the same fingerprint "%v" was already recorded at path "%v", `+
+				`recording a new one at path "%v". `, fingerprint, existingPath, recordName,
+		)})
+	}
+
 	r.records[memoryRecord.Name] = memoryRecord
+	r.recordedFingerprints[fingerprint] = recordName
```

Comment:
I was thinking if we need the two maps....Anyway, won't this `recordedFingerprints` map just get bigger and bigger potentially? I think we should clear it same as the `records` map.

Reply:
oops